sui-execution/latest/sui-move-natives/src/eth_state_proof.rs

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2024 dWallet Labs Ltd.
+// SPDX-License-Identifier: BSD-3-Clause-Clear
+
+use std::collections::VecDeque;
+use std::fs::File;
+use std::io::Read;
+use std::str::FromStr;
+
+use ethers::prelude::EIP1186ProofResponse;
+use ethers::{
+    types::U256,
+    utils::{hex, keccak256},
+};
+use helios::config::networks::Network;
+use helios::consensus::ConsensusStateManager;
+use helios::consensus::{nimbus_rpc::NimbusRpc, Bytes32, ConsensusRpc};
+use helios::dwallet::encode_account;
+use helios::execution;
+use helios::execution::{get_message_storage_slot, verify_proof};
+use helios::prelude::{AggregateUpdates, FinalityUpdate, OptimisticUpdate, Update};
+use helios::types::Address;
+use move_binary_format::errors::{PartialVMError, PartialVMResult};
+use move_binary_format::file_format::StructFieldInformation::Native;
+use move_core_types::{gas_algebra::InternalGas, vm_status::StatusCode};
+use move_vm_runtime::{native_charge_gas_early_exit, native_functions::NativeContext};
+use move_vm_types::{
+    loaded_data::runtime_types::Type,
+    natives::function::NativeResult,
+    pop_arg,
+    values::{Value, Vector},
+};
+use smallvec::smallvec;
+use tracing::log::error;
+
+use crate::object_runtime::ObjectRuntime;
+use crate::NativesCostTable;
+
+/// The value of `True` in the contract's storage map.
+const TRUE_VALUE: u8 = 1;
+
+#[derive(Clone)]
+pub struct EthDWalletCostParams {
+    /// Base cost for invoking the `verify_eth_state` function
+    pub verify_eth_state_cost_base: InternalGas,
+    /// Base cost for invoking the `verify_message_proof` function
+    pub verify_message_proof_cost_base: InternalGas,
+    /// Base cost for invoking the `create_initial_eth_state_data` function
+    pub create_initial_eth_state_data_cost_base: InternalGas,
+}
+
+/***************************************************************************************************
+ * native fun verify_message_proof
+ * Implementation of the Move native function `verify_message_proof(p0: &mut NativeContext, p1: Vec<Type>, p2: VecDeque<Value>) -> PartialVMResult<NativeResult>`
+ * gas cost: verify_message_proof_cost_base   | base cost for function call and fixed operations.
+ **************************************************************************************************/
+pub fn verify_message_proof(
+    context: &mut NativeContext,
+    _ty_args: Vec<Type>,
```

Comment:
```suggestion
```

Reply:
I think this is a part of a native function signature.